app/src/main/java/ai/elimu/appstore/BaseApplication.java

```diff
@@ -77,16 +78,17 @@ protected void log(int priority, String tag, String message, Throwable throwable
         Database db = helper.getWritableDb();
         daoSession = new CustomDaoMaster(db).newSession();
 
-        VersionHelper.updateAppVersion(getApplicationContext());
+        int oldVersionCode = AppPrefs.getAppVersionCode();
+        if (oldVersionCode < 2000010) {
```

Comment:
What if this is the first time the user opens the app? Won't that produce oldVersionCode = 0, and this return `true` here? How can we separate the cases 1) first startup, and 2) upgrade from previously installed version?

Reply:
I already mentioned in previous comment that if we place the logic for deleting APKs here, the APKs will always be deleted whenever we open the app with old version.

In order to separate first startup vs upgrade from prev. installed version, we can place the deletion logic in onUpgrade of CustomDaoMaster because it will be triggered only when there is version upgrade.